### PR TITLE
Fix for ability to activate buttons using right click

### LINKED ISSUE #2565: Buttons are activated by clicking them with right button
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Open https://sdk.ckeditor.com/samples/classic.html
2. Right click any button inside the toolbar.

### Expected result

Nothing happens.

### Actual result

* Button is activated.
* Additionally native context menu could open.

## Other details

* Browser: IE, Edge
* CKEditor version: 4.5.0+

Seems like it's connected with the fact that clicks are handled by `mouseup` event in IEs: https://github.com/ckeditor/ckeditor-dev/blob/0806fb6c214d78caeda6f5b2ef3f729669b4a009/plugins/button/plugin.js#L32

### PR DESCRIPTION
**Pull request by Comandeer**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

I've added additional check if left mouse button was used for click. I also add ability to get info about mouse button from native DOM event in `CKEDITOR.tools.getMouseButton`.

Closes #2565.

### CODE REVIEW

File: tests/plugins/button/button.js [hunk 1]
```diff
@@ -41,5 +52,45 @@ bender.test( {
...
+
+		assert.areSame( 0, spy.callCount );
 	}
 } );
+
+function dispatchMouseEvent( element, type, button ) {
+	var ie8ButtonMap = {
+			0: 1,
+			1: 4,
+			2: 2
```
[the remark this answers is not in the document]
Comment (Comandeer) [reply tests/plugins/button/button.js]: Comments sound like a good idea 👍

File: core/tools.js [hunk 2]
```diff
@@ -1464,17 +1464,18 @@
...
 		getMouseButton: function( evt ) {
-			var evtData = evt.data,
-				domEvent = evtData && evtData.$;
+			var domEvent = evt.data && evt.data.$;
 
-			if ( !( evtData && domEvent ) ) {
-				// Added in case when there's no data available. That's the case in some unit test in built version which
-				// mock event but doesn't put data object.
+			if ( evt instanceof Event ) {
+				domEvent = evt;
```
[the remark this answers is not in the document]
Comment (Comandeer) [reply core/tools.js]: I'd rather keep it the way it is. After inlining we'd have two conditionals (ternary operator and `&&`) in one line.

**Comment by Comandeer** [comment]
The issue with Edge is present also on current `master` branch, so it's a separate bug. What's more you must click toolbar button as the first thing after selecting text, clicking anywhere else first and then on button does not trigger the bug. However I don't think there is much sense in reporting it, as it's a real edge case (no pun intended).

**Comment by engineering-this** [comment]
I could resolve my own suggestion, but only the first one. For the second GH said 'You are not allowed to resolve suggestions'. So I just manually pushed commit 😂

**Comment by Comandeer** [comment]
Fixed failing tests. It was connected with too strict conditional checking if the event is instance of native DOM event. Now the function assumes that passed `evt` option is native DOM event if it doesn't have `data` property.

I've also refactored button's template according to the advice to be more readable.

**Pull request merged**